### PR TITLE
ovl_En_GirlA OK and documented

### PR DESCRIPTION
**Pull request by hensldm**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

As part of this I have also brought over some MACROs from OOT and filled out some more of Z64item.h

### CODE REVIEW

File: include/macros.h [hunk 1]
```diff
@@ -31,28 +32,46 @@
...
+    ((((void)0, gSaveContext.inventory.upgrades) & gUpgradeMasks[upg]) >> gUpgradeShifts[upg])
+
+#define ALL_EQUIP_VALUE(equip) ((gSaveContext.inventory.equipment & gEquipMasks[equip]) >> gEquipShifts[equip])
+#define CUR_EQUIP_VALUE(equip) ((gSaveContext.equips.equipment & gEquipMasks[equip]) >> gEquipShifts[equip])
+#define CUR_UPG_VALUE(upg) ((gSaveContext.inventory.upgrades & gUpgradeMasks[upg]) >> gUpgradeShifts[upg])
+
+#define CAPACITY(upg, value) gUpgradeCapacities[upg][value]
+#define CUR_CAPACITY(upg) CAPACITY(upg, CUR_UPG_VALUE(upg) - 4)
+
+#define SQ(x) ((x) * (x))
```
Comment (Kenix3) [inline include/macros.h:53]: This looks like it's also on line 99

File: src/overlays/actors/ovl_En_GirlA/z_en_girla.h [hunk 2]
```diff
@@ -5,11 +5,61 @@
...
-    /* 0x144 */ char unk_144[0x88];
+    /* 0x144 */ char unk144[0x44];
+    /* 0x188 */ EnGirlAActionFunc actionFunc;
+    /* 0x18C */ s8 objIndex;
+    /* 0x190 */ EnGirlAActionFunc mainActionFunc;
+    /* 0x194 */ s32 isInitialized;
+    /* 0x198 */ u16 choiceTextId;
+    /* 0x19C */ s32 getItemId;
+    /* 0x1A0 */ s16 isOutOfStock;
+    /* 0x1A4 */ void (*boughtFunc)(GlobalContext*, struct EnGirlA*);
```
Comment (Kenix3) [inline src/overlays/actors/ovl_En_GirlA/z_en_girla.h:36]: Why not make a typedef for this function pointer type?

**Review by Kenix3 — CHANGES_REQUESTED** [review]
Couple small things. Good work overall!

File: include/macros.h (hunk 1 above)
Comment (hensldm) [reply include/macros.h:53]: Weird, wonder how that got duplicated.

File: src/overlays/actors/ovl_En_GirlA/z_en_girla.h (hunk 2 above)
Comment (hensldm) [reply src/overlays/actors/ovl_En_GirlA/z_en_girla.h:36]: It was more that these two had the same pointer type as the buy functions, and I couldn't think of a good name to tie all of them together. Perhaps `shopActionFunc`?

File: src/overlays/actors/ovl_En_GirlA/z_en_girla.c [hunk 3]
```diff
@@ -20,90 +62,544 @@ const ActorInit En_GirlA_InitVars = {
...
+}
+
+s32 EnGirlA_CanBuyPotionBlue(GlobalContext* globalCtx, EnGirlA* this) {
+    if (!(gSaveContext.weekEventReg[53] & 8)) {
+        return CANBUY_RESULT_CANT_GET_NOW;
+    }
+    if (!func_80114E90()) {
+        return CANBUY_RESULT_NEED_EMPTY_BOTTLE;
+    }
+    if (!(gSaveContext.weekEventReg[53] & 16)) {
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_GirlA/z_en_girla.c:214]:
```suggestion
    if (!(gSaveContext.weekEventReg[53] & 0x10)) {
```
I think with bitflags (with the exception of single digits), it looks better to have it in hex

File: src/overlays/actors/ovl_En_GirlA/z_en_girla.c [hunk 4]
```diff
@@ -20,90 +62,544 @@ const ActorInit En_GirlA_InitVars = {
...
+    func_801159EC(-globalCtx->msgCtx.unk1206C);
+}
+
+void EnGirlA_BuyMaskAllNight(GlobalContext* globalCtx, EnGirlA* this) {
+    Item_Give(globalCtx, ITEM_MASK_ALL_NIGHT);
+    func_801159EC(-globalCtx->msgCtx.unk1206C);
+}
+
+void EnGirlA_BuyBombBag(GlobalContext* globalCtx, EnGirlA* this) {
+    //! @bug: Bomb Bag paramters in sShopItemEntries are 1 2 3, not 20 21 22
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_GirlA/z_en_girla.c:437]:
```suggestion
    //! @bug: Bomb Bag parameters in sShopItemEntries are 1 2 3, not 20 21 22
```

File: src/overlays/actors/ovl_En_GirlA/z_en_girla.c [hunk 5]
```diff
@@ -20,90 +62,544 @@ const ActorInit En_GirlA_InitVars = {
...
+// Left over from OOT
+s32 EnGirlA_TrySetMaskItemDescription(EnGirlA* this, GlobalContext* globalCtx) {
+    return false;
+}
+
+void EnGirlA_InitalUpdate(EnGirlA* this, GlobalContext* globalCtx) {
+    s16 params = this->actor.params;
+    ShopItemEntry* shopItem = &sShopItemEntries[params];
+
+    if (Object_IsLoaded(&globalCtx->objectCtx, this->objIndex) != 0) {
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_GirlA/z_en_girla.c:540]:
```suggestion
    if (Object_IsLoaded(&globalCtx->objectCtx, this->objIndex)) {
```

**Review by engineer124 — COMMENTED** [review]
Really nice work! Could only find these small things.

**Review by engineer124 — APPROVED** [review]

File: src/overlays/actors/ovl_En_GirlA/z_en_girla.h (hunk 2 above)
Comment (Kenix3) [reply src/overlays/actors/ovl_En_GirlA/z_en_girla.h]: Yes.

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**